src/main/bash/sdkman-install.sh

```diff
@@ -114,65 +114,56 @@ function __sdkman_install_local_version() {
 }
 
 function __sdkman_download() {
-	local candidate version archives_folder
+	local candidate version
 
 	candidate="$1"
 	version="$2"
 
-	archives_folder="${SDKMAN_DIR}/archives"
 	metadata_folder="${SDKMAN_DIR}/var/metadata"
 	mkdir -p ${metadata_folder}
 		
-	if [ ! -f "${archives_folder}/${candidate}-${version}.zip" ]; then
-		local platform_parameter="$(echo $SDKMAN_PLATFORM | tr '[:upper:]' '[:lower:]')"
-		local download_url="${SDKMAN_CANDIDATES_API}/broker/download/${candidate}/${version}/${platform_parameter}"
-		local base_name="${candidate}-${version}"
-		local zip_archive_target="${SDKMAN_DIR}/archives/${base_name}.zip"
-		local tmp_headers_file="${SDKMAN_DIR}/tmp/${base_name}.headers.tmp"
-		local headers_file="${metadata_folder}/${base_name}.headers"
-
-		# pre-installation hook: implements function __sdkman_pre_installation_hook
-		local pre_installation_hook="${SDKMAN_DIR}/tmp/hook_pre_${candidate}_${version}.sh"
-		__sdkman_echo_debug "Get pre-installation hook: ${SDKMAN_CANDIDATES_API}/hooks/pre/${candidate}/${version}/${platform_parameter}"
-		__sdkman_secure_curl "${SDKMAN_CANDIDATES_API}/hooks/pre/${candidate}/${version}/${platform_parameter}" >| "$pre_installation_hook"
-		__sdkman_echo_debug "Copy remote pre-installation hook: $pre_installation_hook"
-		source "$pre_installation_hook"
-		__sdkman_pre_installation_hook || return 1
-		__sdkman_echo_debug "Completed pre-installation hook..."
-
-		export local binary_input="${SDKMAN_DIR}/tmp/${base_name}.bin"
-		export local zip_output="${SDKMAN_DIR}/tmp/$base_name.zip"
+	local platform_parameter="$(echo $SDKMAN_PLATFORM | tr '[:upper:]' '[:lower:]')"
+	local download_url="${SDKMAN_CANDIDATES_API}/broker/download/${candidate}/${version}/${platform_parameter}"
+	local base_name="${candidate}-${version}"
+	local tmp_headers_file="${SDKMAN_DIR}/tmp/${base_name}.headers.tmp"
+	local headers_file="${metadata_folder}/${base_name}.headers"
+
+	# pre-installation hook: implements function __sdkman_pre_installation_hook
+	local pre_installation_hook="${SDKMAN_DIR}/tmp/hook_pre_${candidate}_${version}.sh"
+	__sdkman_echo_debug "Get pre-installation hook: ${SDKMAN_CANDIDATES_API}/hooks/pre/${candidate}/${version}/${platform_parameter}"
+	__sdkman_secure_curl "${SDKMAN_CANDIDATES_API}/hooks/pre/${candidate}/${version}/${platform_parameter}" >| "$pre_installation_hook"
+	__sdkman_echo_debug "Copy remote pre-installation hook: $pre_installation_hook"
+	source "$pre_installation_hook"
+	__sdkman_pre_installation_hook || return 1
+	__sdkman_echo_debug "Completed pre-installation hook..."
+
+	export local binary_input="${SDKMAN_DIR}/tmp/${base_name}.bin"
+	export local zip_output="${SDKMAN_DIR}/tmp/${base_name}.zip"
```

Comment:
The target file here is a zip, but only because the post-install hook transforms the tarball into a zip. The hook will need to determine if it is a tarball or zip based on the content header, then place it here either as `.tar.gz` or `.zip`. Then we will need to deal with it appropriately.